apps/api/src/app/layouts/usecases/create-layout-change/create-layout-change.use-case.ts

```diff
@@ -0,0 +1,33 @@
+import { LayoutEntity, LayoutRepository } from '@novu/dal';
+import { ChangeEntityTypeEnum } from '@novu/shared';
+import { Injectable } from '@nestjs/common';
+
+import { CreateLayoutChangeCommand } from './create-layout-change.command';
+
+import { LayoutDto } from '../../dtos/layout.dto';
+import { ChannelTypeEnum, IEmailBlock } from '../../types';
+import { CreateChange, CreateChangeCommand } from '../../../change/usecases';
+
+@Injectable()
+export class CreateLayoutChangeUseCase {
+  constructor(private createChange: CreateChange, private layoutRepository: LayoutRepository) {}
+
+  async execute(command: CreateLayoutChangeCommand): Promise<void> {
+    const item = await this.layoutRepository.findById(command.layoutId, command.environmentId);
```

Comment:
Generic use case to be reused in all the Layout use cases where we need to track a change.

Reply:
In the case of create change after delete-layout, this will not find the deleted one

Reply:
How do we handle deletions? With the soft delete should be able to pick it up. Don't know how mongo-delete works 100%.

Reply:
I made a suggestion in this [pr](https://github.com/novuhq/novu/pull/2574) for a find soft deleted layouts. Let me know what you think :)

Reply:
So the change creation for a deletion would need to use that new method?

Reply:
Maybe if he doesn't find the item like this, then do the findDeleted